tests/scipy_spatial_test.py

```diff
@@ -0,0 +1,309 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+
+from jax._src import test_util as jtu
+from jax.scipy.spatial.transform import Rotation as jsp_Rotation
+from scipy.spatial.transform import Rotation as osp_Rotation
+from jax.scipy.spatial.transform import Slerp as jsp_Slerp
+from scipy.spatial.transform import Slerp as osp_Slerp
+
+import jax.numpy as jnp
+from jax.config import config
+
+config.parse_flags_with_absl()
+
+float_dtypes = jtu.dtypes.floating
+real_dtypes = float_dtypes + jtu.dtypes.integer + jtu.dtypes.boolean
+
+num_samples = 2
+
+class LaxBackedScipySpatialTransformTests(jtu.JaxTestCase):
+  """Tests for LAX-backed scipy.spatial implementations"""
+
+  @jtu.sample_product(
+    dtype=float_dtypes,
+    shape=[(4,), (num_samples, 4)],
+    vector_shape=[(3,), (num_samples, 3)],
+    inverse=[True, False],
+  )
+  def testRotationApply(self, shape, vector_shape, dtype, inverse):
+    rng = jtu.rand_default(self.rng())
+    args_maker = lambda: (rng(shape, dtype), rng(vector_shape, dtype),)
+    jnp_fn = lambda q, v: jsp_Rotation.from_quat(q).apply(v, inverse=inverse)
+    np_fn = lambda q, v: osp_Rotation.from_quat(q).apply(v, inverse=inverse)
+    self._CheckAgainstNumpy(np_fn, jnp_fn, args_maker, check_dtypes=False,
```

Comment:
Why do we need `check_dtypes=False` here and throughout the tests? How are JAX's returned dtypes diverging from NumPy's?

Reply:
Changed all of these to `check_dtypes=True`. I had just copied this from somewhere else without thinking about it too much.